jtravis/src/main/java/fr/inria/spirals/jtravis/parsers/MavenLogParser.java

```diff
@@ -12,80 +14,114 @@
+            this.globalResults.setErrored(this.globalResults.getErrored() + Integer.parseInt(mvnTestNumberMatcher.group(3)));
+            res.setErrored(Integer.parseInt(mvnTestNumberMatcher.group(3)));
 
             if (nbMatch == 5) {
-                this.skippingTests += Integer.parseInt(mvnTestNumberMatcher.group(5));
+                this.globalResults.setSkipping(this.globalResults.getSkipping() + Integer.parseInt(mvnTestNumberMatcher.group(5)));
+                res.setSkipping(Integer.parseInt(mvnTestNumberMatcher.group(5)));
             }
+
+            res.setPassing(res.getRunning() - (res.getFailing()+res.getErrored()+res.getSkipping()));
+            this.detailedResults.add(res);
+            return  true;
         }
+        else
+            return false;
     }
 
     @Override
     public TestsInformation parseLog(TravisFold outOfFold) {
-        boolean beginTestHead = false;
-        boolean inTestHead = false;
         boolean inTestBlock = false;
+        boolean inTestResults = false;
+        this.globalResults = new TestsInformation();
+        this.detailedResults = new ArrayList<TestsInformation>();
 
-        Pattern mvnLinePattern = Pattern.compile(MVN_LINE_PATTERN);
         Pattern mvnTestHeadPattern = Pattern.compile(MVN_TESTS_PATTERN);
-
+        Pattern mvnTestResultsPattern = Pattern.compile(MVN_RESULTS_PATTERN);
         for (String s : outOfFold.getContent()) {
-            Matcher mvnLineMatcher = mvnLinePattern.matcher(s);
-
-            if (!beginTestHead && !inTestHead && mvnLineMatcher.matches()) {
-                beginTestHead = true;
-                inTestBlock = false;
+            Matcher mvnTestResultsMatcher = mvnTestResultsPattern.matcher(s);
+            Matcher mvnTestHeadMatcher = mvnTestHeadPattern.matcher(s);
+            if (!inTestBlock && mvnTestHeadMatcher.matches()) {
+                inTestBlock = true;
                 continue;
             }
-
-            if (beginTestHead && !inTestHead && mvnLineMatcher.matches()) {
-                beginTestHead = false;
-                inTestBlock = false;
+            if (inTestBlock && mvnTestResultsMatcher.matches()) {
+                inTestResults = true;
                 continue;
             }
 
-            if (beginTestHead && inTestHead && mvnLineMatcher.matches()) {
-                beginTestHead = false;
-                inTestHead = false;
-                inTestBlock = true;
-                continue;
+            if (inTestResults) {
+                if(this.parseTestLine(s)){
+                    inTestBlock = false;
+                    inTestResults = false;
+                }
+
             }
+        }
 
-            Matcher mvnTestHeadMatcher = mvnTestHeadPattern.matcher(s);
+        this.computePassingTests();
+        return this.globalResults;
+    }
+
+    @Override
+    public List<TestsInformation> parseDetailedLog(TravisFold outOfFold) {
+        boolean inTestBlock = false;
```

Comment:
Actually here the code seems pretty similar than for `parseLog`, and it seems useless to compute twice the logs if you want first the global results and then the detailed one. 
I propose you add a boolean somewhere to know if the logs had been parsed, and if it's the case you return the results. Then you should have a private method `parseLog()` which returns nothing but only do the parsing job, and the returning method will only test the boolean call the private method if needed, and return the results.